packages/renderer/src/stores/kubernetes-contexts-state.ts

```diff
@@ -166,6 +166,31 @@ export const kubernetesCurrentContextNodesFiltered = derived(
   ([$searchPattern, $nodes]) => $nodes.filter(node => findMatchInLeaves(node, $searchPattern.toLowerCase())),
 );
 
+// Pods
+
+export const kubernetesCurrentContextPods = readable<KubernetesObject[]>([], set => {
+  window
+    .kubernetesRegisterGetCurrentContextResources('pods')
+    .then(value => set(value))
+    .catch((err: unknown) => console.log('Error registering Kubernetes pods', err));
+  window.events?.receive('kubernetes-current-context-pods-update', (value: unknown) => {
+    set(value as KubernetesObject[]);
```

Comment:
issue: need a check before the cast ?
(if the event is not the one expected)

Reply:
Is there anything we can do to enforce event-specific type safety / allow typed values?

In general I agree we should, or have more defensive code, but this same pattern is used 8 other places within this file, as well as several other events, and many commands that have arguments. I can open a new issue to cover this, but IMHO for this PR it is best to follow the existing pattern.

Reply:
#9866 opened as follow-up.